astroquery/cadc/core.py

```diff
@@ -135,19 +153,29 @@ def login(self, user=None, password=None, certificate_file=None):
         certificate : str, required if user is None
             path to certificate to use with logging in
 
-        Notes
-        -----
-        This will soon be deprecated as it does not make sense to login with
-        certificates.
         """
         # start with a new session
-        pyvo.dal.tap.s = requests.Session()
-        if certificate_file:
-            pyvo.dal.tap.s.cert = certificate_file
-        elif not user or not password:
+        if not isinstance(self.cadctap._session, (requests.Session,
+                                                  authsession.AuthSession)):
+            raise TypeError('Cannot login with user provided session that is '
+                            'not an pyvo.authsession.AuthSession or '
+                            'requests.Session')
+        if not certificate_file and not (user and password):
             raise AttributeError('login credentials missing (user/password '
                                  'or certificate)')
-        else:
+        if certificate_file:
+            if isinstance(self.cadctap._session, authsession.AuthSession):
+                self.cadctap._session.credentials.\
```

Comment:
we change the suggested line lenght in astropy to 100, so this should be fine. (no need to change now, I'm saying more as an info)